provider.go

```diff
@@ -88,8 +90,122 @@ func (EnvProvider) Priority() int {
 	return 2
 }
 
-func (EnvProvider) FormatKey(key string) (env string) {
-	env = strings.ToUpper(key)
+// FormatEnvKey transforms configuration keys to environment variable format.
+// Examples: "database.url" -> "DATABASE_URL", "api-key" -> "API_KEY"
+func FormatEnvKey(key string) string {
+	env := strings.ToUpper(key)
 	env = strings.Replace(env, ".", "_", -1)
 	return strings.Replace(env, "-", "_", -1)
 }
+
+func (EnvProvider) FormatKey(key string) (env string) {
+	return FormatEnvKey(key)
+}
+
+// A Provider that implements the repository.Provider interface for dotenv files.
+type DotenvProvider struct {
+	vars map[string]string
+}
+
+// NewDotenvProvider creates a provider that loads environment variables from a .env file
+// in the current directory. If the file doesn't exist, the provider will be empty but still functional.
+func NewDotenvProvider() *DotenvProvider {
+	return NewDotenvProviderWithPath(".env")
+}
+
+// NewDotenvProviderWithPath creates a provider that loads environment variables from the specified .env file.
+// If the file doesn't exist, the provider will be empty but still functional.
+func NewDotenvProviderWithPath(path string) *DotenvProvider {
+	p := &DotenvProvider{
+		vars: make(map[string]string),
+	}
+
+	// Make path absolute to avoid issues with working directory changes
+	if !filepath.IsAbs(path) {
+		if abs, err := filepath.Abs(path); err == nil {
+			path = abs
+		}
+	}
+
+	p.loadFile(path)
+	return p
+}
+
+// loadFile loads variables from the specified dotenv file.
+func (p *DotenvProvider) loadFile(path string) {
+	file, err := os.Open(path)
+	if err != nil {
+		// File doesn't exist or can't be opened, but that's okay
+		return
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// Skip empty lines and comments
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// Parse KEY=VALUE format
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		key := strings.TrimSpace(parts[0])
+		value := strings.TrimSpace(parts[1])
+
+		// Parse quoted values with basic unquoting
```

Comment:
Please consider this

```suggestion
		// Remove surrounding quotes if present
		if strings.HasPrefix(value, `"`) && strings.HasSuffix(value, `"`) || strings.HasPrefix(value, `'`) && strings.HasSuffix(value, `'`) {
			value = value[1 : len(value)-1]
		}
```